opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -17,78 +17,49 @@
+from opentelemetry.sdk.metrics.view import ViewData, ViewManager
 from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.util import get_dict_as_key
 from opentelemetry.sdk.util.instrumentation import InstrumentationInfo
 
 logger = logging.getLogger(__name__)
 
 
-def get_labels_as_key(labels: Dict[str, str]) -> Tuple[Tuple[str, str]]:
-    """Gets a list of labels that can be used as a key in a dictionary."""
-    return tuple(sorted(labels.items()))
-
-
 class BaseBoundInstrument:
     """Class containing common behavior for all bound metric instruments.
 
     Bound metric instruments are responsible for operating on data for metric
     instruments for a specific set of labels.
 
     Args:
-        value_type: The type of values for this bound instrument (int, float).
-        enabled: True if the originating instrument is enabled.
-        aggregator: The aggregator for this bound metric instrument. Will
-            handle aggregation upon updates and checkpointing of values for
-            exporting.
+        labels: A set of labels as keys that bind this metric instrument.
+        metric: The metric that created this bound instrument.
     """
 
     def __init__(
         self,
-        value_type: Type[metrics_api.ValueT],
-        enabled: bool,
-        aggregator: Aggregator,
+        labels: Tuple[Tuple[str, str]],
+        metric: metrics_api.MetricT,
     ):
-        self.value_type = value_type
-        self.enabled = enabled
-        self.aggregator = aggregator
-        self._ref_count = 0
-        self._ref_count_lock = threading.Lock()
+        self._labels = labels
+        self._metric = metric
 
     def _validate_update(self, value: metrics_api.ValueT) -> bool:
-        if not self.enabled:
+        if not self._metric.enabled:
             return False
-        if not isinstance(value, self.value_type):
+        if not isinstance(value, self._metric.value_type):
             logger.warning(
-                "Invalid value passed for %s.", self.value_type.__name__
+                "Invalid value passed for %s.",
+                self._metric.value_type.__name__
             )
             return False
         return True
 
     def update(self, value: metrics_api.ValueT):
-        self.aggregator.update(value)
-
-    def release(self):
-        self.decrease_ref_count()
-
-    def decrease_ref_count(self):
-        with self._ref_count_lock:
-            self._ref_count -= 1
-
-    def increase_ref_count(self):
-        with self._ref_count_lock:
-            self._ref_count += 1
-
-    def ref_count(self):
-        with self._ref_count_lock:
-            return self._ref_count
-
-    def __repr__(self):
-        return '{}(data="{}")'.format(
-            type(self).__name__, self.aggregator.current
-        )
-
+        # The view manager handles all updates to aggregators
+        self._metric.meter.view_manager.record(self._metric, self._labels, value)
```

Comment:
BoundInstruments are meant to offer highest performance to users, as it avoids look up to which timeseries the value should go. The original code met this requirement, as it had a reference to the aggregator and just needed to call aggregator.update.

With the new code, the boundinstrument is calling view_manager.record. The view_manager has to do a lookup first with the labels, to find out which timeseries this update should go into. This would be defeating the purpose of boundinstruments.

I don't have solution to this - but i believe this is a p0 scenario to solve. As discussed in our offline conversation,  I'd add a comment in view otep as well.

Reply:
Good point, this does look like something we have to solve to make views viable.

@lzchen and I talked about this offline yesterday and we may be able to avoid the lookup cost here (and a lot of complexity elsewhere) if we don't support adding and removing views on the fly, e.g. by replacing `View.__init__` with `ViewManager.configure(views: list[View])`. In that case we could create each bound instrument with a set of aggregators without having to worry about keeping that set up to date.

Reply:
@c24t Yes! I also did some prototyping in .NET. 
If Views config is fixed , then it simplifies a lot of things.

Reply:
@cijothomas 
Updated the implementation to include high performance of bound instruments (aggregators are initialized upon boundinstrument creation based off of configured views at point of creation).